src/com/facebook/buck/ide/intellij/BaseIjModuleRule.java

```diff
@@ -118,18 +149,84 @@ private void addDepsAndFolder(
     }
   }
 
+  private void addDepsAndFolder(
+      IJFolderFactory folderFactory,
+      DependencyType dependencyType,
+      TargetNode<T, ?> targetNode,
+      boolean wantsPackagePrefix,
+      ModuleBuildContext context) {
+    addDepsAndFolderWithFolderInputIndexFilter(
+        folderFactory, dependencyType, targetNode, wantsPackagePrefix, context, null);
+  }
+
+  protected void addDepsAndSourcesWithFolderInputIndexFilter(
+      TargetNode<T, ?> targetNode,
+      boolean wantsPackagePrefix,
+      ModuleBuildContext context,
+      @Nullable Predicate<? super Map.Entry<Path, Path>> folderInputIndexFilter) {
+    addDepsAndFolderWithFolderInputIndexFilter(
+        SourceFolder.FACTORY,
+        DependencyType.PROD,
+        targetNode,
+        wantsPackagePrefix,
+        context,
+        folderInputIndexFilter);
+  }
+
   protected void addDepsAndSources(
       TargetNode<T, ?> targetNode, boolean wantsPackagePrefix, ModuleBuildContext context) {
     addDepsAndFolder(
         SourceFolder.FACTORY, DependencyType.PROD, targetNode, wantsPackagePrefix, context);
   }
 
+  protected void addDepsAndTestSourcesWithFolderInputIndexFilter(
+      TargetNode<T, ?> targetNode,
+      boolean wantsPackagePrefix,
+      ModuleBuildContext context,
+      @Nullable Predicate<? super Map.Entry<Path, Path>> folderInputIndexFilter) {
+    addDepsAndFolderWithFolderInputIndexFilter(
+        TestFolder.FACTORY,
+        DependencyType.TEST,
+        targetNode,
+        wantsPackagePrefix,
+        context,
+        folderInputIndexFilter);
+  }
+
   protected void addDepsAndTestSources(
       TargetNode<T, ?> targetNode, boolean wantsPackagePrefix, ModuleBuildContext context) {
     addDepsAndFolder(
         TestFolder.FACTORY, DependencyType.TEST, targetNode, wantsPackagePrefix, context);
   }
 
+  // If there is no resources_root, then we use the java src_roots option from .buckconfig, so
```

Comment:
Rewrote the comment to give a bit more context for why the method exists and has the given return type. If it still isn't too helpful, then I'll just remove it?